lam4doc/adapters/sparql_adapter.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/python3
+
+# sparql_adapter.py
+# Date:  20/12/2020
+# Author: Laurentiu Mandru
+# Email: mclaurentiu79@gmail.com
+
+"""
+Adapter for the SPARQL wrapper library
+"""
+import logging
+from abc import ABC, abstractmethod
+from json import JSONDecodeError
+from urllib.parse import urljoin
+
+from requests.auth import HTTPBasicAuth
+from requests_toolbelt import MultipartEncoder
+
+from lam4doc import config
+from lam4doc.adapters.helpers import get_file_format
+from lam4doc.config import LAM_LOGGER
+
+logger = logging.getLogger(LAM_LOGGER)
+
+
+class FusekiException(Exception):
```

Comment:
this seems to be repeated several times. How could we write a more reusable code? 
shall we consider some new library?

Reply:
in the future we could extract the code from multiple projects into a Fuseki related library. I'm not sure it will work for other triple stores, though. Fuseki does not seem to adhere to SPARQL graph store HTTP protocol.